### PR TITLE
added some contributions guide

### PR DESCRIPTION
**Pull request by RohitRathore1**

### CODE REVIEW

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:30]:
```suggestion
* find a suitable definition in `src/` that the docstring applies to;
```

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:33]:
```suggestion
* add the name of the definition to the `@docs` code block. For example, with a docstring added to a function `bar`.
```

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:32]:
```suggestion
* find a suitable `@docs` code block in the `docs/src/lib/functions.md` file where you would like the docstring to appear;
```

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:41]:
```suggestion
you would add the name `bar` to a `@docs` block in `docs/src/lib/functions.md`
```

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:50]: is it indeed? I thought you need to run `julia make.jl` from the `/docs` directory?

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:56]:
```suggestion
Examples written within docstrings can be used as test cases known as *doctests* by annotating code blocks with `jldoctest`.
```

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:63]:
```suggestion
A doctest needs to match an interactive REPL including the `julia>` prompt. To run doctests you need to run `make -C doc doctest=true` from the root directory.

It is recommended to add the header `# Examples` above the doctests.
```

Comment (bkamins) [reply CONTRIBUTING.md:63]: is the `make -C doc doctest=true` ineede the command to run?

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:116]:
```suggestion
* Avoid working from the `main` branch of your fork, creating a new branch will make it easier if DataFrame.jl `main` branch changes and you need to update your pull request.
```

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:117]:
```suggestion
```

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:118]: there is no need to do it - we will squash the PR when merging it anyway. The request is rather: run tests of your code before sending any commit to GitHub. Only push changes when the tests of the change are passing locally.

The point is that if you send e.g. 10 commits to GitHub in one push operation it is not a problem as CI will be run only once then.

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:120]:
```suggestion
* If any conflicts arise due to changes in DataFrame.jl `main` branch, prefer updating your pull request branch with `git rebase` (rather than `git merge`), since the latter will introduce a merge commit that might confuse GitHub when displaying the diff of your PR, which makes your changes more difficult to review. Alternatively use conflict resolution tool available at GitHub.
```

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:122]:
```suggestion
* Please try to use descriptive commit messages to simplify the review process.
```

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:126]: It is not needed. Maintainers get notified of all changes in the repository. However, what is useful is writing a short message after a sequence of changes is made summarizing what has changed and that the PR is ready for a review.

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:132]: I would rather write:
1. please make sure you follow the code formatting guidelines when submitting a PR
2. in particular do not remove whitespace in parts of code that you are not editing as this makes reviewing the PR harder (I regularly do code cleanup anyway)
3. (and note that in some doctests actually whitespace might be relevant and it should not be removed) - this is the case in multiline (triple-quoted) strings. Therefore using single quoted strings with an explicit `"\n"` and joined with `*` across multiple lines should be done in such cases.

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:128]: I understand your intention here, but I am not sure I fully understand the recommendation. I.e. when I kit `y` key nothing happens.

**Comment by bkamins** [comment]
Ah - and you might add that if a PR is not finished yet and SHOULD NOT be reviewed yet then it should be opened as DRAFT (in this way I will know I can ignore such PR until it is made non-draft or the author asks for a review).

**Comment by bkamins** [comment]
Also maybe mention somewhere (if it is not mentioned yet) that all PRs and issues should be made against the `main` branch not against the current release as this is one of the most common problems.

File: CONTRIBUTING.md
Comment (RohitRathore1) [reply CONTRIBUTING.md:50]: Both will work but `julia make.jl` is better. So, i will change it

File: CONTRIBUTING.md
Comment (RohitRathore1) [reply CONTRIBUTING.md:63]: If someone doesn't want to change `doctest=true` in make.jl

**Comment by RohitRathore1** [comment]
> Also maybe mention somewhere (if it is not mentioned yet) that all PRs and issues should be made against the `main` branch not against the current release as this is one of the most common problems.

Ohh, okay. I am making the required changes. Thanks for reviewing

File: CONTRIBUTING.md
Comment (bkamins) [reply CONTRIBUTING.md:63]: but we have `doctest=true` in make.jl - right?

File: CONTRIBUTING.md
Comment (RohitRathore1) [reply CONTRIBUTING.md:128]: It happens when you select some code of any file and then hit the `y` key: let's say I select the first 4 lines of [core.jl](https://github.com/JuliaData/DataFrames.jl/tree/main/src/join) then after selection url will look like this: https://github.com/JuliaData/DataFrames.jl/blob/a671a3b86537a253d461ff54b87ae2c26e4e8a04/src/join/core.jl
and before selection:
https://github.com/JuliaData/DataFrames.jl/tree/main/src/join

**Comment by bkamins** [comment]
Code organization comments before I review again:
1. Can you please add line breaks to very long lines?
2. Ca you please use punctuation at the end of lines (in particular when making bulleted lists)

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:123]: wrong alignment here and below (one space is missing in the front)

**Comment by RohitRathore1** [comment]
Made required changes.

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:20]:
```suggestion
be found by clicking on the `source` link that appears below each docstring in the documentation.
```

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:25]: As discussed - why do you write `make dos`? I would assume `julia --project=docs docs/make.jl` run from the top-level folder of the package? But maybe I am missing something here?

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:23]: maybe add step 1. create a new branch?

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:32]:
```suggestion
* find a suitable function or type definition in `src/` that the docstring
  will be most applicable to;
```

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:35]:
```suggestion
  would like the docstring to appear; (if the docstring is added
  to an object that is not exported add it to `docs/src/lib/internals.md`);
```

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:56]: maybe add below an example of preferred style of docstring, i.e.:

1. signature
2. what function does
3. description of its arguments
4. additional details if needed
5. examples

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:69]: as commented `doctest=true` is already set so I think there is no need to mention that.

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:126]:
```suggestion
* All PRs and issues should be opened against the `main` branch not against the current release;
```

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:130]:
```suggestion
* Run tests of your code before sending any commit to GitHub. Only push changes when 
  the tests of the change are passing locally. In particular note that it is not a problem
  if you send several commits in one push command to GitHub as CI will be run only once then;
```

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:137]:
```suggestion
* Maintainers get notified of all changes made on GitHub. However, what is useful is writing a short
```

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:141]:
```suggestion
  while viewing code on GitHub to reload the page with a URL that includes the specific commit that 
```

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:142]:
```suggestion
  you're viewing. That way any lines of code that you refer to will still be correct in the future, even 
```

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:143]:
```suggestion
  if additional commits are pushed to the branch you are reviewing;
```

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:151]:
```suggestion
* If a PR is not finished yet and should not be reviewed yet then it should be opened as DRAFT 
```

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:152]:
```suggestion
  (in this way maintainers will know that they can ignore such PR until it is made non-draft or the author asks for a review).
```

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:150]:
```suggestion
* Please make sure you follow the code formatting guidelines when submitting a PR;
  However, preferably do not modify parts of code that you are not editing as this makes
  reviewing the PR harder (it is better to open a separate maintenance PR
  if e.g. code layout can be improved);
```

File: CONTRIBUTING.md
Comment (RohitRathore1) [reply CONTRIBUTING.md:56]: You want me to add an example here or I have to just write these lines?

Comment (bkamins) [reply CONTRIBUTING.md:56]: whatever you would feel would be more useful for a new contributor.

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:29]:
```suggestion
* a preferred structure of docstring is:
```

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:34]:
```suggestion
  + list of accepted signatures;
  + a short information what function does;
  + description of its arguments;
  + additional details if needed;
  + examples.
```

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:70]:
```suggestion
blocks with `jldoctest`:
```

File: CONTRIBUTING.md
Comment (bkamins) [inline CONTRIBUTING.md:153]:
```suggestion
  Also preferably do not modify parts of code that you are not editing as this makes
```

File: CONTRIBUTING.md
Comment (nalimilan) [inline CONTRIBUTING.md:134]:
```suggestion
  easier if DataFrames.jl `main` branch changes and you need to update your pull request;
```

File: CONTRIBUTING.md
Comment (nalimilan) [inline CONTRIBUTING.md:139]:
```suggestion
* If any conflicts arise due to changes in DataFrames.jl `main` branch, prefer updating your pull
```

**Comment by RohitRathore1** [comment]
Fixed errors which were requested by @nalimilan

**Comment by bkamins** [comment]
For a reference: you can assume that @nalimilan noticed that. We need to wait for his approval. In general expect that things take several days when working on PRs (so this means that it is good to have several things in parallel in the pipeline if you can spend more time on working on the package).

**Review by nalimilan — APPROVED** [review]
Fine with me if it's OK for you @bkamins.

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**
